tpm2/structures.go

```diff
@@ -1021,10 +1021,37 @@ type AuthCommand struct {
 	Auth       tpmutil.U16Bytes
 }
 
+// BDigest implements the structure of TPM2B_Digest as described in Part 2 Structures, Page 97
+type BDigest struct {
+	Size   uint16
+	Digest []byte
+}
+
 // TPMLDigest represents the TPML_Digest structure
 // It is used to convey a list of digest values.
 //This type is used in TPM2_PolicyOR() and in TPM2_PCR_Read()
 type TPMLDigest struct {
 	Count   uint32
-	Digests []tpmutil.U16Bytes
+	Digests []BDigest
+}
+
+// ToBytes converts the TPMLDigest structure into a byte slice
+func (list *TPMLDigest) ToBytes() ([]byte, error) {
+	if uint32(len(list.Digests)) != list.Count {
```

Comment:
If you implement a custom Encode() function for this struct, why not eliminate the Count field altogether? That way, there isn't an inconsistency that could exist.

Reply:
Yes, but the field exists because the spec says so. My point is: you read the spec, you see the code and there is no magic happening in between....for example the whole concept of U16Bytes and U32Bytes is a nice solution programming wise, but it is a huge pain in the ass to handle and it obfuscates parts of the specification.

Reply:
I'd like to discuss further: what specifically is painful about U16Bytes/U32Bytes and family? A TSS's job is to make it easier to write TPM commands, otherwise we'd all be assembling raw TPM commands byte by byte. If you walk me through why every user should have to copy len(someSlice) into a field to properly call a command, maybe we can find the source of the general problem with the API that we need to solve.

Re: your general point that the TSS and the TPM API should be well (and completely) mapped to each other - well heard, and I think go-tpm has some room to grow in that aspect. But, IMHO, U16Bytes is not the root cause of the dissonance here.

Reply:
The tpmutil.Pack() function can't handle slices of U16Bytes, and my inability to change that, brought me to that statement mostly. Was not my best statement, nor most elaborate. I see the point.